source/core/DebugPoint.cpp

```diff
@@ -10,14 +10,19 @@
 
 namespace Vireo {
     //------------------------------------------------------------
-    // Debug node which will check if breakpoint is set or not
+    // Debug node which will check if breakpoint is set or not and do return cildsac for VI pause state
     VIREO_FUNCTION_SIGNATURE1(DebugPoint, StringRef)
     {
         if (THREAD_EXEC()->debuggingContext->GetDebugPointState(_Param(0)->MakeSubStringAlias())) {
 #if kVireoOS_emscripten
+            THREAD_EXEC()->_viPaused = true;
+            THREAD_EXEC()->ExecuteTillNextStopPoint();
             jsDebuggingContextDebugPointInterrupt(_Param(0));
 #endif
         }
+        if (THREAD_EXEC()->_viPaused) {
+            return _this;
```

Comment:
Shouldn't this return CulDeSac?

Reply:
Hmm.. I am guessing you did this since you need the _savePc to be enqueued. Then the check in ExecuteTillNextStopPoint is wrong since it checks for a CulDeSac instruction.

Reply:
Reverted changes